src/plugins/intel_gpu/src/runtime/ocl/ocl_memory.cpp

```diff
@@ -110,6 +110,15 @@ event::ptr gpu_buffer::copy_from(stream& stream, const void* host_ptr) {
     return ev;
 }
 
+event::ptr gpu_buffer::copy_to(stream& stream, void* host_ptr) {
+    auto& cl_stream = downcast<ocl_stream>(stream);
+    auto ev = stream.create_base_event();
+    cl::Event ev_ocl = std::dynamic_pointer_cast<ocl_event>(ev)->get();
+    cl_stream.get_cl_queue().enqueueReadBuffer(_buffer, false, 0, size(), host_ptr, nullptr, &ev_ocl);
```

Comment:
Does it always enqueue read buffer regardless the memory is host or device?

Reply:
Hmm, I think this method is applicable only for OpenCL buffer -> host buffer copy, for OpenCL buffer -> OpenCL buffer enqueueCopyBuffer will be called
Do I understand correctly what do you mean?